lib/plugins/aws/provider/awsProvider.js

```diff
@@ -237,6 +237,56 @@ class AwsProvider {
             },
           },
         },
+        resources: {
+          properties: {
+            // Not replicating the full JSON schema from https://s3.amazonaws.com/cfn-resource-specifications-us-east-1-prod/schemas/2.15.0/all-spec.json
+            // as that gets into the specifics for each resource type.
+            //
+            // The only required attribute is `Type`; `Properties` and other common attributes are optional.
+            // See also https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resources-section-structure.html
+            Resources: {
+              type: 'object',
+              patternProperties: {
+                '^[a-zA-Z0-9]{1,255}$': {
+                  type: 'object',
+                  properties: {
+                    Type: { type: 'string' },
```

Comment:
It'll probably be good to add definition for `Properties` (just `type: 'object') and all properties listed here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-product-attribute-reference.html (just sanity checks).

Ideally if we define all those properties (aside of `Type`)  as reusable definition. For that I'd add support for provider to define its own definitions here:

https://github.com/serverless/serverless/blob/5f85543d8e47a9afaec6811a57e1235e82949550/lib/classes/ConfigSchemaHandler/index.js#L172

So they're added to this object: https://github.com/serverless/serverless/blob/5f85543d8e47a9afaec6811a57e1235e82949550/lib/configSchema.js#L116

With this move I would also define `awsArn` and `awsCfImport` in scope of `defineProvider` call (move out from core schema).

Having that I would reuse this definition here, in form:

```javascript
{
  properties: { $ref: '#/definitions/awsResource' },
  additionalProperties: {
    Type: { type: "string: }
  }
```

and reuse it also in context of `resources.extensions`

Reply:
I've put in a first attempt at a smoke-check definition for `Properties`, `CreationPolicy`, `DeletionPolicy`, `DependsOn`, `Metadata`, `UpdatePolicy`, and `UpdateReplacePolicy`. I'm not sure I understand why you want these to be reusable definitions?

Note that the supported properties for `resources.extensions` are different from the supported properties for `resources`, so we can't re-use the type definition for `resources` and `resources.extensions`. `Type` is explicitly not supported for `resources.extensions` (see #7352 and specifically [this switch statement](https://github.com/serverless/serverless/blob/5f85543d8e47a9afaec6811a57e1235e82949550/lib/plugins/aws/package/lib/mergeCustomProviderResources.js#L36)).

I can look into adding support for extending `definitions` and moving `awsArn` and `awsCfImport`.

Reply:
> Note that the supported properties for resources.extensions are different from the supported properties for resources,

Both `resources.Resources` and `resource.extensions` extend CloudFormation resources, why do you assume that supported properties are different?

Reply:
@medikoo `resources.Resources` allows you to add new resources or extend existing resources; `resource.extensions` explicitly only allows you to extend existing resources. When I wrote #7352 at your request, the implementation intentionally disallowed specifying `Type` in `resources.extensions` for this reason.

Reply:
Yes, afaik the only difference between them is that `Type` is not supported in context of `resources.extensions`. 

That's also the reason why I suggested to exclude `Type` in generalization. Afaik generalization as I proposed can work well for both, or do I miss something?